### PR TITLE
VM issue: fix #749, #753, #575

### LINKED ISSUE #749: [docs] Do not include individual names in docs
```
~ az vm extension set -h

[...]

Examples
    Add a new linux user
        az vm extension set -n VMAccessForLinux --publisher Microsoft.OSTCExtensions --version 1.4
        --vm-name myvm --resource-group yugangw --private-config '{"username":"user1",
        "ssh_key":"ssh_rsa ..."}'
``

I'm a bit against this stuff. The docs can be a little bit more generic/formal.

### PR DESCRIPTION
**Pull request by yugangw-msft**

### CODE REVIEW

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py [hunk 1]
```diff
@@ -377,13 +377,15 @@ def delete_application(client, identifier):
...
-    except ValueError:
-        result = list(client.list(filter="identifierUris/any(s:s eq '{}')".format(identifier)))
+    result = list(client.list(filter="identifierUris/any(s:s eq '{}')".format(identifier)))
+    if not result:
+        try:
+            uuid.UUID(identifier)
+            #it is either app id or object id, let us verify
+            result = list(client.list(filter="appId eq '{}'".format(identifier)))
+        except ValueError:
+            raise CLIError("Application '{}' doesn't exist".format(identifier))
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py]: this is to expose correct error message on AD graph exceptions

**Comment by yugangw-msft** [comment]
/CC: @burtbiel, @tjprescott , @derekbekoe, @johanste
Several trivial bug fixes

**Comment by BurtBiel** [comment]
:shipit:

**Comment by derekbekoe** [comment]
:shipit:

**Comment by tjprescott** [comment]
:shipit:

**Pull request merged**